hail/python/hailtop/batch/batch_pool_executor.py

```diff
@@ -478,12 +478,15 @@ async def async_cancel(self):
         ``True`` is returned if the job is cancelled. ``False`` is returned if
         the job has already completed.
         """
+
         if self.fetch_coro.cancelled():
             return False
+
         if self.fetch_coro.done():
             # retrieve any exceptions raised
             self.fetch_coro.result()
             return False
+
         await self.batch.cancel()
```

Comment:
We should remove this as well because cancelling the fetch_coro (on the next line) will cancel the batch.